src/test/java/Pages/BasePage.java

```diff
@@ -0,0 +1,35 @@
+package Pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.By;
+
+import java.time.Duration;
+
+public class BasePage {
+    public static WebDriver driver = null;
+    public static WebDriverWait wait = null;
+    public static Actions actions = null;
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
    protected Actions actions;
    protected WebDriverWait wait;
    protected WebDriver driver;
```

Reply:
When I'm making variables protected all files getting errors

Reply:
> When I'm making variables protected all files getting errors

Hello

Please, re-submit homework 22 for review

Reply:
Sure. I just changed selectors to private. For Actions , WebDriverWait '
    and WebDriver I couldn't make them protected because it makes errors

Reply:
I resubmited it.

Reply:
> Sure. I just changed selectors to private. For Actions , WebDriverWait ' and WebDriver I couldn't make them protected because it makes errors

1
```
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class BasePage {
    protected WebDriver driver;
    protected WebDriverWait wait;
    protected Actions actions;

    public BasePage(WebDriver givenDriver) {
        driver = givenDriver;
        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        actions = new Actions(driver);
    }

    public WebElement findElement(By locator) {
        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
    }

    protected void doubleClick(By locator) {
        WebElement element = findElement(locator);
        actions.doubleClick(element).perform();
    }
}
```
![image](https://github.com/Test-Pro-Bootcamp/QA-Automation-Framework-Template/assets/104761482/40edc197-a019-4c83-9d82-0333091e9a2f)

2

```
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.annotations.BeforeSuite;
import org.testng.annotations.Test;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Parameters;


import java.time.Duration;

@Test
public class BaseTest {
    public static WebDriver driver = null;
    @BeforeSuite
    public static void setupClass() {
        WebDriverManager.chromedriver().setup();
    }
    @BeforeMethod
    @Parameters({"BaseUrl"})
    public void launchBrowser(String BaseUrl) {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--remote-allow-origins=*");
        driver = new ChromeDriver(options);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));

        String url = BaseUrl;
        driver.navigate().to(url);


    }

}
```

![image](https://github.com/Test-Pro-Bootcamp/QA-Automation-Framework-Template/assets/104761482/c453b283-bfa3-4ebf-b85a-9eaa2c4fde36)